soh/CMakeLists.txt

```diff
@@ -284,23 +284,67 @@ if (CMAKE_SYSTEM_NAME STREQUAL "Windows")
 	set_target_properties(${PROJECT_NAME} PROPERTIES MSVC_RUNTIME_LIBRARY ${MSVC_RUNTIME_LIBRARY_STR})
 endif()
 
+include(FetchContent)
+
 ################################################################################
 # apclientpp + dependencies
 ################################################################################
 add_compile_definitions(ASIO_STANDALONE)
 
-target_include_directories(${PROJECT_NAME} PRIVATE
-    ${CMAKE_CURRENT_SOURCE_DIR}/../subprojects/asio/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/../subprojects/websocketpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/../subprojects/wswrap/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/../subprojects/valijson/include
-    ${CMAKE_CURRENT_SOURCE_DIR}/../subprojects/apclientpp
-)
+if(BUILD_REMOTE_CONTROL)
+    if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+        FetchContent_Declare(
+            valijson
+            GIT_REPOSITORY https://github.com/tristanpenman/valijson.git
+            GIT_TAG v1.0.6
+        )
+        FetchContent_MakeAvailable(valijson)
+        target_include_directories(${PROJECT_NAME} PRIVATE ${valijson_SOURCE_DIR}/include)
+    else()
+        find_package(valijson REQUIRED)
+    endif()
+
+    FetchContent_Declare(
+            asio
+            GIT_REPOSITORY https://github.com/chriskohlhoff/asio.git
+            GIT_TAG asio-1-30-2
+    )
+    FetchContent_MakeAvailable(asio)
+    target_include_directories(${PROJECT_NAME} PRIVATE ${asio_SOURCE_DIR}/asio/include)
+
+    # websocketpp has been temporarily removed from vcpkg, see
+    # https://github.com/microsoft/vcpkg/pull/42678#issuecomment-2914451436
+    if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+        FetchContent_Declare(
+            websocketpp
+            GIT_REPOSITORY https://github.com/zaphoyd/websocketpp.git
+            GIT_TAG 0.8.2
+        )
+        FetchContent_Populate(websocketpp)
+        target_include_directories(${PROJECT_NAME} PRIVATE ${websocketpp_SOURCE_DIR})
+    else()
+        find_package(websocketpp REQUIRED)
+    endif()
+
+    FetchContent_Declare(
+            wswrap
+            GIT_REPOSITORY https://github.com/black-sliver/wswrap.git
+            GIT_TAG 47438193ec50427ee28aadf294ba57baefd9f3f1
+    )
+    FetchContent_MakeAvailable(wswrap)
+    target_include_directories(${PROJECT_NAME} PRIVATE ${wswrap_SOURCE_DIR}/include)
+
+    FetchContent_Declare(
+            apclientpp
+            GIT_REPOSITORY https://github.com/black-sliver/apclientpp.git
+            GIT_TAG 65638b7479f6894eda172e603cffa79762c0ddc1
```

Comment:
fetchcontent was failing to fetch using the sha the submodule had

the submodule was using [`a5b7b96b6bfbdc7f2831796a4e126bcc09c35da1`](https://github.com/black-sliver/apclientpp/tree/a5b7b96b6bfbdc7f2831796a4e126bcc09c35da1) which doesn't belong to a branch on the repo. the commit message for that sha is "bump version to 0.6.0" which matches [`65638b7479f6894eda172e603cffa79762c0ddc1`](https://github.com/black-sliver/apclientpp/tree/65638b7479f6894eda172e603cffa79762c0ddc1) which exists on main. when doing a comparison to main (https://github.com/black-sliver/apclientpp/compare/main...a5b7b96b6bfbdc7f2831796a4e126bcc09c35da1) i didn't see any commit messages that didn't exist on main, so i assume it just had something to do with a rebase or something at some point.